pkg/utils/k8sutil.go

```diff
@@ -921,6 +945,11 @@ func ensureFuncDeployment(client kubernetes.Interface, funcObj *spec.Function, o
 	//prepare init-container for custom runtime
 	initContainer := v1.Container{}
 	if funcObj.Spec.Deps != "" {
```

Comment:
There should be a validation for non-empty runtime here. Otherwise, there will be error if the function is deployed with custom runtime and dependency.